### DISCUSSION TITLE
Mobile App

### DISCUSSION THREAD

**Discussion by Olisteve** [jgraph/drawio#2090] [Ideas]
Will there be a mobile version? I can see myself needing it on my tablet. Thanks for the great work

**Comment by Peeeet** [comment]
It almost is kinda usable for me as-is on Android.  The trick is opening the "Outline" Dialog for panning around the document since the scroll bars don't work and there is no "Pan" hand tool like in other software.

Honestly all that needs to be implemented is multi-touch pan / zoom on the canvas and I think it would work fine.

**Comment by layer8braincourt** [comment]
An app for iPad with pencil support would be great - ideally with support for all storage option, e.g. local, Office 365 or Atlassian Cloud.

**Comment by jwlehane** [comment]
I know this is outdated, but the real issue is that the help files on the web page make it sound like touch to drag and two finger touch to drag and zoom is enabled, and it is not. Thank you for the tip on opening The outline dialogue. That made it useful on a tablet for me

**Reply by alderg** [reply]
What device and operating system are you using?

**Reply by jwlehane** [reply]
Google Pixel tablet android v15

**Comment by alderg** [comment]
Unfortunately, we don't have Android v15 to test. It does work on Android v11. What version do you get on https://app.diagrams.net/clear.html?

**Reply by jwlehane** [reply]
I tried the link above that you provided. I get the same result and I shared a video exampleClearing cache, version 26.0.4
https://github.com/user-attachments/assets/c786a67f-3a9e-47e6-ba2a-aaa08754298f

**Comment by jwlehane** [comment]
Clearing Cached version 26.0.4... OK

On Mon, Dec 30, 2024 at 2:37 AM Gaudenz Alder ***@***.***>
wrote:

> Unfortunately, we don't have Android v15 to test. It does work on Android
> v11. What version do you get on https://app.diagrams.net/clear.html?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jgraph/drawio/discussions/2090#discussioncomment-11694887>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7I6CGOCEPUCD4K7GQ7VQL2IDZ43AVCNFSM6AAAAABUJBO2KCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRZGQ4DQNY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Johnny LeHane
703.626.2055 (mobile)
***@***.***

<http://www.linkedin.com/in/jwlehane>
 ~  *The future is already here — it's just not very evenly distributed. * ~
William Gibson

**Comment by shoffmeister** [comment]
Alas, canvas drag and pinch gestures do not seem to work for me on a fast 12.5" Android 14 tablet, with the very latest Chrome browser and the very latest drawio 26.1.1.

Whatever I do, it translates always to selection actions. This happens for both finger and high precision pen input (Lenovo Precision Pen)

This behaviour is counter to https://www.drawio.com/blog/touch-screen-diagrams It also seems as if there is no difference between long touch and short touch - perhaps that is the key?

For reference https://patrickhlauke.github.io/touch/tracker/multi-touch-tracker-pointer-hud.html shows great touch Event Support on this tablet in general and https://patrickhlauke.github.io/touch/pinch-zoom-rotate/ for specific actions

Finally, the demo at https://www.jointjs.com/demos/touch-gestures works as I would expect drawio to behave.

**Comment by alderg** [comment]
Works for me on an Android Galaxy Tab A9 with draw.io 26.1.1, Android 14, One UI 6.1 and Chrome 134.0.6998.95 (tried canvas drag, pinch and rubberband selection). Rubberband selection starts on touch devices if you **tap and hold** for >350ms.

**Reply by shoffmeister** [reply]
I just tried this with draw.io 26.1.3, on Chrome (134.0.6998.135), on Android 14 (Samsung One-UI 6.1), on a Samsung Galaxy A52s. 

There everything works as expected.

So I retried this with draw.io 26.1.3, on Chrome (134.0.6998.95) on the "Xiaoxin Pad Pro 12.7 (2025) == TB375FC" (ZUI 16, Chinese device in English language) approximately equal to "Lenovo Idea Tab Pro == TB373FU".

As before, rubberband selection is immediate and does not kicked in delayed by 350ms. And that apparently prevents drag/pinch from working.

So, in a somewhat naïve attempt to debug this, I discovered https://eruda.liriliri.io/ ("console for mobile browsers") and - even more importantly - the sibling project https://github.com/liriliri/eruda-android: "This simple browser like app loads eruda automatically on every page, useful for testing and debugging android system webview"

What can I say - draw.io _works_ in that app (in the Android system webview)?!

Additional quick browser checks:
* Microsoft Edge (Android) - same as Chrome
* Firefox 136.0.2 - rubberband + drag/pinch work, but moving objects on the canvas does not work

And here comes the kicker: 
* draw.io works in Chrome - *if "Desktop site" is toggled off.*
* draw.io works in Microsoft Edge - *if "Mobile site" is toggled on.*

Not sure whether draw.io can be amended to work in both Desktop and Mobile mode?

**Reply by shoffmeister** [reply]
And switching to mobile in Chrome and then installing the "app" will transport that setting over, make the app work.

**Reply by alderg** [reply]
What do you get on https://www.whatismybrowser.com/detect/what-is-my-user-agent/ for those tablets?

**Reply by shoffmeister** [reply]
Chrome 134.0.6998.135 on "Lenovo Xiaoxin Pad Pro 12.7" (TB375FC), Android 14, ZUI 16.0.20.091

Desktop site on:
```
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/134.0.0.0 Safari/537.36
```

Desktop site off:
```
Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/134.0.0.0 Safari/537.36
```

**Comment by alderg** [comment]
Do you have access to the browser console in the Xiaoxin Pad Pro? If yes, try `mxClient.IS_ANDROID = true;`
Also, what do you get for `mxClient.IS_TOUCH` and `mxClient.IS_POINTER`?

**Comment by ppczouz** [comment]
Reviving this...

I have the same issue with inability to pan and zoom not possible with 2 fingers. (Edit: they work if I switch to Mobile Site)
But my biggest gripe is that I cannot seem to find a way to turn off the immediate type feature... any shape I touch will trigger the soft keyboard. how can I limit that behavior to when in text edit mode (such as after a double click)?

I'm running v29.6.4 on Chrome on Android 14.
Hardware is TCL NXTPAPER 11 Plus

I bought the tablet specifically to run draw.io

**Comment by alderg** [comment]
Thanks for the report. We believe both issues (pan/zoom not working and soft keyboard on tap) are caused by Chrome's "Request Desktop Site" mode changing the user agent so that the app no longer detects the device as Android.

Could you open the browser console on https://app.diagrams.net and paste the following snippet, then share the output here? Please do this **twice** — once with "Desktop site" on (your normal setup) and once with it off:

```js
JSON.stringify({
  IS_ANDROID: mxClient.IS_ANDROID,
  IS_TOUCH: mxClient.IS_TOUCH,
  IS_POINTER: mxClient.IS_POINTER,
  IS_LINUX: mxClient.IS_LINUX,
  IS_GC: mxClient.IS_GC,
  touchStyle: Graph.touchStyle,
  maxTouchPoints: navigator.maxTouchPoints,
  userAgent: navigator.userAgent
}, null, 2)
```

This will help us confirm the root cause and work on a fix.

**Comment by alderg** [comment]
We've added a fix for this. In the next release you can test with the URL parameter `android=1`:

`https://app.diagrams.net/?android=1`

This forces Android detection when Chrome's "Request Desktop Site" changes the user agent. It should fix both the pan/zoom issue and the soft keyboard appearing on cell tap.

We've also added a general fix using `maxTouchPoints` to detect touch-capable devices even when the user agent says Linux, which should help without the parameter in most cases. Please let us know if this resolves the issues.

**Reply by ppczouz** [reply]
I confirm that this works perfectly in v29.6.5 using the query /?android=1 
Panning, Zooming, no immediate edit, and edit with double click all works.
This works on both desktop site and mobile site.
Amazing work!

**Reply by alderg** [reply]
What do you get on https://www.whatismybrowser.com/detect/what-is-my-user-agent/ for your tablet? And https://github.com/jgraph/drawio/discussions/2090#discussioncomment-16260642
